src/providers/BasicPoolsProvider.tsx

```diff
@@ -66,6 +82,24 @@ export type BasicPool = {
 } & SwapInfo
 export type BasicPools = { [poolName: string]: BasicPool | undefined } | null // indexed by name, which is unique in the Registry
 
+type SwapStorage = [
```

Comment:
protip, TS also lets you do `ReturnType<typeof PoolRegistry.swapStorage>` so you can get rid of this 

https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
Let me look into this. Could potentially cause problems since PoolRegistry.getSwapStorage() returns a Promise.

Reply:
Refer to ethcall.d.ts Call for how to get see the result of the Promise